napari/plugins/_npe2.py

```diff
@@ -233,9 +233,16 @@ def iter_manifests(
 
 def widget_iterator() -> Iterator[Tuple[str, Tuple[str, Sequence[str]]]]:
     # eg ('dock', ('my_plugin', ('My widget', MyWidget)))
+    # Make a dict of display_name matching plugin_name
+    plugin_names = {}
+    for plugin in pm.iter_manifests():
+        plugin_names[plugin.name] = plugin.display_name
+
```

Comment:
I think this can be avoided with the use of `pm.get_manifest()`. See my next comment.